virtcontainers/sandbox.go

```diff
@@ -1462,6 +1469,20 @@ func (s *Sandbox) setSandboxPid(pid int) error {
 	return s.storage.storeSandboxResource(s.id, stateFileType, s.state)
 }
 
+// setCgroupPath sets the cgroup for the sandbox, sandbox's processes including
+// the hypervisor are placed there.
+func (s *Sandbox) setCgroupPath(c *Container) error {
+	if s.state.Pid != c.process.Pid {
```

Comment:
For shimv2, there is no kata shim process for a container, and the container's process pid is set to "-1", the sandbox's state Pid is also set "-1", thus here we cannot assert whether a container is sandbox container by test "s.state.Pid != c.process.Pid".  it's better to use  the line https://github.com/kata-containers/runtime/blob/1c27ab79d9ce76202aade8b2124a003363419fc3/virtcontainers/container.go#L764 to assert.  Or you can move calling the setCgroupPath() to there.